Improved Raspberry Pi Pico support

LinuxJedi opened the pull request:
Fixes and improvements for RP2040 and added RP2350 support:

* Made UART `printf` output optional
* Made WiFi support optional
* Added compile support for RP2350
* Re-wrote README
* Fixed compile issues

Requires wolfSSL/wolfssl#8153

LinuxJedi commented:
Addressed comments indicated in a Slack conversation.

dgarske commented on RPi-Pico/config/user_settings.h:
```diff
@@ -33,8 +33,11 @@ extern "C"
+#include <time.h>
 #define TARGET_EMBEDDED
 
+#define WOLFSSL_RPIPICO
```
Please add `WOLFSSL_IGNORE_FILE_WARN` to suppress the warning.

LinuxJedi replied:
Done

dgarske commented on RPi-Pico/CMakeLists.txt:
```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.13)
 # Pull in Pico and FreeRTOS
 include(pico_sdk_import.cmake)
-include(pico_extras_import_optional.cmake)
+#include(pico_extras_import_optional.cmake)
```
```diff --git a/RPi-Pico/CMakeLists.txt b/RPi-Pico/CMakeLists.txt
index 9af73bd..b3b2868 100644
--- a/RPi-Pico/CMakeLists.txt
+++ b/RPi-Pico/CMakeLists.txt
@@ -1,7 +1,8 @@
 cmake_minimum_required(VERSION 3.13)

 # Pull in Pico and FreeRTOS
-include(pico_sdk_import.cmake)
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}")
+include($ENV{PICO_SDK_PATH}/external/pico_sdk_import.cmake)
 #include(pico_extras_import_optional.cmake)
 #include($ENV{FREERTOS_KERNEL_PATH}/portable/ThirdParty/GCC/RP2040/FreeRTOS_Kernel_import.cmake)```

LinuxJedi replied:
The `set` was only needed for the other thing we tried, not needed for that final line, so I've only done the final one and updated the README accordingly.

dgarske reviewed (changes requested):
Cannot get past this error with the blink:

```
[ 41%] Built target benchmark
[ 41%] Building C object CMakeFiles/Wifi.dir/src/blink.c.obj
/home/davidgarske/GitHub/wolfssl-examples/RPi-Pico/src/blink.c:23:10: fatal error: pico/cyw43_arch.h: No such file or directory
   23 | #include "pico/cyw43_arch.h"
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/Wifi.dir/build.make:76: CMakeFiles/Wifi.dir/src/blink.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:1840: CMakeFiles/Wifi.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

LinuxJedi commented:
OK, lots of improvements in this morning's update:

* Added the compile fixes we discussed
* Fixed the issue you had with WiFi compiling (see below)
* Added WiFi setup options to README and descriptions in CMake
* Added a prerequisites section
* Added a section for uploading to the Pico
* Added a section for terminal output
* Specified that you can use any `PICO_BOARD` and pointed to the list

## WiFi compiling issues

So, `USE_WIFI` was on by default, but unless `PICO_BOARD` was set to a board that had a CYW43 chip, the include directories / linking was not setup for WiFi. Now:

* `USE_WIFI` is now off by default
* If you try to use `USE_WIFI` without a `PICO_BOARD` that has a CYW43 CMake will error with "YYou can only set USE_WIFI when a PICO_BOARD with wifi is used"

LinuxJedi commented:
Another push because I quickly added RISC-V mode for the RP2350 too.

dgarske reviewed (approved):
Worked amazing! 
wolfCrypt test/bench both run and pass.
Tested on MacOS with Intel and Rpi5.
Ran on RP2040
Thank you
